packages/host/app/resources/auto-attached-card.ts

```diff
@@ -82,13 +110,19 @@ export class AutoAttachment extends Resource<Args> {
 export function getAutoAttachment(
   parent: object,
   args: {
+    submode: () => Submode;
```

Comment:
if this is a thunk that means this resource responds to the user switching submodes. which means that you should add a test where there is an attachment made in code mode and the user switches to interact mode or vice versa, and you should confirm that the behavior is consistent when the submode changes.